apps/gateway/lib/gateway/controllers/curse_of_mirra/item_controller.ex

```diff
@@ -20,13 +21,17 @@ defmodule Gateway.Controllers.CurseOfMirra.ItemController do
 
   # This is a placeholder and will be developed properly in https://github.com/lambdaclass/mirra_backend/issues/638
   def buy(conn, params) do
-    with {:ok, item_template_id} <-
-           Items.get_purchasable_template_id_by_name_and_game_id(
+    with {:ok, item_template} <-
+           Items.get_purchasable_template_by_name_and_game_id(
```

Comment:
I'm not a fan of this function, it seems too specific. Why not just checking that `item.purchasable?` is true in the `with` statement. Then we can use a less specific `GameBackend.Items.get_item_template_by_name_and_game`, which is more similar to our other functions in `GameBackend`

Reply:
Oh, seems like it wasn't introduced by your PR hehe. I think since you're touching it anyways it wouldn't hurt to refactor it here while we're at it anyways.

Reply:
Nice catch! 
purchasable? field was a placeholder, I already replaced it here -> https://github.com/lambdaclass/mirra_backend/pull/651/files

Reply:
Oh that's great!